newapi/ooniapi/measurements.py

```diff
@@ -1473,3 +1497,133 @@ def get_torsf_stats() -> Response:
 
     except Exception as e:
         return jsonify({"v": 0, "error": str(e)})
+
+
+# # measurement feedback
+
+from ooniapi.auth import role_required, get_account_id, jerror
+from ooniapi.database import insert_click
+
+
+"""
+CREATE TABLE msmt_feedback
+(
+    `measurement_uid` String,
+    `account_id` String,
+    `status` String,
+    `comment` String,
+    `update_time` DateTime64(3) MATERIALIZED now64()
+)
+ENGINE = ReplacingMergeTree
+ORDER BY (measurement_uid, account_id)
+SETTINGS index_granularity = 4
+"""
+
+valid_feedback_status = [
+    "blocked",
+    "blocked.blockpage",
+    "blocked.blockpage.http",
+    "blocked.blockpage.dns",
+    "blocked.blockpage.server_side",
+    "blocked.blockpage.server_side.captcha",
+    "blocked.dns",
+    "blocked.dns.inconsistent",
+    "blocked.dns.nxdomain",
+    "blocked.tcp",
+    "blocked.tls",
+    "ok",
+    "ok.unreachable",
+    "ok.broken",
+    "ok.parked",
+]
+
+
+@api_msm_blueprint.route("/v1/measurement_feedback")
+@metrics.timer("get_msmt_feedback")
+@role_required(["admin", "user"])
+def get_msmt_feedback() -> Response:
+    """Get measurement for the curred logged user for a given measurement
+    ---
+    produces:
+      - application/json
+    consumes:
+      - application/json
+    parameters:
+      - in: body
+        required: true
+        schema:
+          type: object
+          properties:
+            measurement_uid:
+              type: string
+              description: Measurement ID
+    responses:
+      200:
+        description: status and comment or empty JSON
+    """
+
+    def jparam(name):
+        return request.json.get(name, "").strip()
+
+    log = current_app.logger
+    account_id = get_account_id()
+    measurement_uid = jparam("measurement_uid")
+
+    query = """SELECT status, comment FROM msmt_feedback FINAL
+    WHERE account_id = :account_id AND measurement_uid = :measurement_uid
+    LIMIT 1"""
+    qp = dict(account_id=account_id, measurement_uid=measurement_uid)
+    row = query_click_one_row(sql.text(query), qp)
+    if row:
+        return cachedjson("0s", **row)
+
+    return cachedjson("0s")
+
+
+@api_msm_blueprint.route("/v1/submit_measurement_feedback", methods=["POST"])
```

Comment:
I would suggest placing this under the same path as the `get_msmt_feedback` (i.e. `/measurement_feedback`), but with the POST method instead of get.

Reply:
A dedicated entry point is slightly more descriptive "submit_" and easier to search in logs.

Reply:
It's considered an anti-pattern to have verbs inside of API endpoints, see:
* https://stackoverflow.blog/2020/03/02/best-practices-for-rest-api-design/#h-use-nouns-instead-of-verbs-in-endpoint-paths
* https://swagger.io/resources/articles/best-practices-in-api-design/#describe-resource-functionality-with-http-methods-4
* https://hevodata.com/learn/rest-api-best-practices/#6
etc.

Reply:
Updated